MSSQL/OMOPLoader.sql

```diff
@@ -22,108 +22,175 @@
+----------------------------------------------------------------------------------------------------------
 IF  EXISTS (SELECT * FROM sys.synonyms WHERE name = N'pcornet_diag') DROP SYNONYM pcornet_diag
 IF  EXISTS (SELECT * FROM sys.synonyms WHERE name = N'pcornet_demo') DROP SYNONYM pcornet_demo
 IF  EXISTS (SELECT * FROM sys.synonyms WHERE name = N'pcornet_proc') DROP SYNONYM pcornet_proc
-IF  EXISTS (SELECT * FROM sys.synonyms WHERE name = N'pcornet_lab') DROP SYNONYM pcornet_lab
 IF  EXISTS (SELECT * FROM sys.synonyms WHERE name = N'pcornet_med') DROP SYNONYM pcornet_med
 IF  EXISTS (SELECT * FROM sys.synonyms WHERE name = N'pcornet_vital') DROP SYNONYM pcornet_vital
 IF  EXISTS (SELECT * FROM sys.synonyms WHERE name = N'pcornet_enc') DROP SYNONYM pcornet_enc
-IF OBJECTPROPERTY (object_id('dbo.getDataMartID'), 'IsScalarFunction') = 1 DROP function getDataMartID
-IF OBJECTPROPERTY (object_id('dbo.getDataMartName'), 'IsScalarFunction') = 1 DROP function getDataMartName
-IF OBJECTPROPERTY (object_id('dbo.getDataMartPlatform'), 'IsScalarFunction') = 1 DROP function getDataMartPlatform
-IF  EXISTS (SELECT * FROM sys.views WHERE name = N'i2o_ontology_lab') DROP VIEW i2o_ontology_lab
-IF  EXISTS (SELECT * FROM sys.views WHERE name = N'i2o_ontology_drug') DROP VIEW i2o_ontology_drug
-GO
-
--- This table needs to be created before the synonyms
-IF  EXISTS (SELECT * FROM sys.objects WHERE object_id = OBJECT_ID(N'[dbo].[i2b2patient_list]') AND type in (N'U'))
-DROP TABLE [dbo].[i2b2patient_list]
-GO
-CREATE TABLE [dbo].[i2b2patient_list]  ( 
-	[patient_num]	int NOT NULL 
-	)
-GO
+IF  EXISTS (SELECT * FROM sys.synonyms WHERE name = N'pcornet_enroll') DROP SYNONYM pcornet_enroll
+IF  EXISTS (SELECT * FROM sys.views WHERE object_id = OBJECT_ID(N'i2o_ontology_lab')) DROP VIEW i2o_ontology_lab
+--IF  EXISTS (SELECT * FROM sys.views WHERE object_id = OBJECT_ID(N'i2o_ontology_drug')) DROP VIEW i2o_ontology_drug
 
--- You will almost certainly need to edit your database name
--- Synonyms for dimension tables
--- NOTE If this synonym doesn't work, run pcornet_prep and try again
-create synonym i2b2visit for i2b2demodata..visit_dimension
-GO 
-create view i2b2patient as select * from i2b2demodata..patient_dimension where patient_num in (select patient_num from i2b2patient_list)
-GO
-create synonym i2b2fact for  i2b2demodata..observation_fact    
-GO
-create synonym i2b2concept for  i2b2demodata..concept_dimension  
-GO
-create synonym provider_dimension for i2b2demodata..provider_dimension
-GO
-
--- You will almost certainly need to edit your database name
--- Synonyms for ontology dimensions and loyalty cohort summary
--- The synonyms in comments have identical names to the tables - 
--- you will only need to edit and uncomment if your tables have
--- names other than these
-
--- These are the new ontology-indepdent i2o-2020 views
-create view i2o_ontology_lab as (select * from i2b2stub..pcornet_lab where i_stddomain='LOINC')

```

Comment:
Your table-based approach for config doesn't support these WHERE constraints. Then again everything should work ok without them, it just would speed things up a bit if you put your entire ontology in one table. What do you think?

Reply:
As discussed earlier excluding the WHERE constraints here and having them including in the Measuremnts Procedure is simplier